third_party/terraform/tests/resource_cloudiot_device_registry_update_test.go

```diff
@@ -0,0 +1,162 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccCloudIoTRegistryBasic(registryName),
+			},
+			{
+				ResourceName:      "google_cloudiot_registry.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckCloudIoTRegistryDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_cloudiot_registry" {
+			continue
+		}
+		config := testAccProvider.Meta().(*Config)
+		registry, _ := config.clientCloudIoT.Projects.Locations.Registries.Get(rs.Primary.ID).Do()
+		if registry != nil {
+			return fmt.Errorf("Registry still present")
+		}
+	}
+	return nil
+}
+
+func testAccCloudIoTRegistryBasic(registryName string) string {
+	return fmt.Sprintf(`
+
+resource "google_project_service" "cloud-iot-apis" {
+  service = "cloudiot.googleapis.com"
+
+  disable_dependent_services = true
+}
+
+resource "google_cloudiot_registry" "%s" {
+  name = "%s"
+
+  depends_on = [
+    google_project_service.cloud-iot-apis
+  ]
+}
+`, registryName, registryName)
+}
+
+func testAccCloudIoTRegistryExtended(registryName string) string {
+	return fmt.Sprintf(`
+
+resource "google_project_service" "cloud-iot-apis" {
+  service = "cloudiot.googleapis.com"
+
+  disable_dependent_services = true
+}
+
+resource "google_project_service" "pubsub-apis" {
+  service = "pubsub.googleapis.com"
+
+  disable_dependent_services = true
+}
+
+resource "google_pubsub_topic" "default-devicestatus" {
+  name = "psregistry-test-devicestatus-%s"
+
+  depends_on = [
+    google_project_service.pubsub-apis
+  ]
+}
+
+resource "google_pubsub_topic" "default-telemetry" {
+  name = "psregistry-test-telemetry-%s"
+
+  depends_on = [
+    google_project_service.pubsub-apis
+  ]
+}
+
+resource "google_pubsub_topic" "additional-telemetry" {
+  name = "psregistry-additional-test-telemetry-%s"
+
+  depends_on = [
+    google_project_service.pubsub-apis
+  ]
+}
+
+resource "google_cloudiot_registry" "%s" {
+  name     = "%s"
+
+  depends_on = [
+    google_pubsub_topic.default-telemetry,
+    google_pubsub_topic.additional-telemetry
+  ]
+
+  event_notification_configs {
+    pubsub_topic_name = google_pubsub_topic.additional-telemetry.id
+    subfolder_matches = "test/directory"
+  }
+
+  event_notification_configs {
+    pubsub_topic_name = google_pubsub_topic.default-telemetry.id
+    subfolder_matches = ""
+  }
+
+  state_notification_config = {
+    pubsub_topic_name = google_pubsub_topic.default-devicestatus.id
+  }
+
+  mqtt_config = {
+    mqtt_enabled_state = "MQTT_DISABLED"
+  }
+
+  http_config = {
+    http_enabled_state = "HTTP_DISABLED"
+  }
+
+  log_level = "INFO"
+
+  credentials {
+    public_key_certificate = {
+      format      = "X509_CERTIFICATE_PEM"
+      certificate = file("test-fixtures/rsa_cert.pem")
+    }
+  }
+}
+`, acctest.RandString(10), acctest.RandString(10), acctest.RandString(10), registryName, registryName)
```

Comment:
can you create a new random variable in the test using `randString(t, 10)` and then pass it here as a parameter instead? we're doing some test overhauls and that's one of the things the  new framework needs

Reply:
done